### ISSUE TITLE
(version 12.0.2) #delete_all generates invalid SQL when used with STI

### ISSUE THREAD

**Issue by seandilda** [composite-primary-keys/composite_primary_keys#529]
AR+composite_primary_keys is generating invalid SQL when I call #delete_all on a model using composite keys and STI

My models look like:

```ruby
class DataSource::Record < ApplicationRecord
  self.primary_keys = :data_source_id, :key
end

class DataSource::Record::Vulnerability < DataSource::Record
end
```

When I call `DataSource::Record::Vulnerability.delete_all`, I receive the following error:
```
irb(main):002:0> DataSource::Record::Vulnerability.delete_all
   (0.6ms)  SET NAMES utf8,  @@SESSION.sql_mode = 'TRADITIONAL',  @@SESSION.sql_auto_is_null = 0, @@SESSION.wait_timeout = 2147483
  DataSource::Record::Vulnerability Destroy (0.7ms)  DELETE FROM `data_source_records` WHERE (data_source_records.data_source_id, data_source_records.key) IN (SELECT data_source_id,key FROM (SELECT DISTINCT `data_source_records`.`data_source_id`, `data_source_records`.`key` FROM `data_source_records` WHERE `data_source_records`.`type` = 'DataSource::Record::Vulnerability') __active_record_temp)
Traceback (most recent call last):
        1: from (irb):2
ActiveRecord::StatementInvalid (Mysql2::Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'key FROM (SELECT DISTINCT `data_source_records`.`data_source_id`, `data_source_r' at line 1)
irb(main):003:0> 
```

Not only is the SQL invalid, its much more complex than needed.  It should be able to call something along the lines of:
```sql
DELETE FROM `data_source_records` WHERE `data_source_records`.`type` = 'DataSource::Record::Vulnerability';
```

I am using:
rails 6.0.3.4
composite_primary_keys 12.0.2
ruby 2.6.6p146
MySQL 8.0.20

**Comment by codeodor** [comment]
@cfis I'm not sure, but seems like this might be related to https://github.com/composite-primary-keys/composite_primary_keys/pull/504

**Comment by cfis** [comment]
This should be fixed by 1e238902e5a5df97eb344926869d5af6ccda3787.
Yes, the generated SQL is complex (although a bit less now). But that's because delete_all can be called on a join, thus the safest way to do this is via a subquery (Rails does that too, but does have a check for the simpler case).
